modules/perception/obstacle/camera/visualizer/glfw_fusion_viewer.cc

```diff
@@ -1310,30 +1326,81 @@ bool GLFWFusionViewer::draw_lane_objects_image(cv::Mat* image_mat) {
       }
     }
 
+    lane_object_color = lane_map_colors[lane_objects_->at(k).newSpatial];
+
+    bool use_new_post_ = false;
     // Besides the fitted polynomial curves we draw lane markers as well
-    for (auto p = lane_objects_->at(k).image_pos.begin();
+    // Upper-left points in image coord
+    if (!use_new_post_) {
+      for (auto p = lane_objects_->at(k).image_pos.begin();
          p != lane_objects_->at(k).image_pos.end(); ++p) {
-      cv::circle(*image_mat,
-                 cv::Point(static_cast<int>(p->x()), static_cast<int>(p->y())),
-                 4, lane_object_color, -1);
-    }
-
-    // draw polynomial curve
-    float img_y_start =
+         cv::circle(*image_mat,
+          cv::Point(static_cast<int>(p->x()), static_cast<int>(p->y())),
+          3, cv::Scalar(0, 0, 0), -1);
+      }
+      // draw polynomial curve
+      // Upper-left points in image coord
+      float img_y_start =
         static_cast<float>(lane_objects_->at(k).img_curve.x_start);
-    float img_y_end = static_cast<float>(lane_objects_->at(k).img_curve.x_end);
-    float start = std::min(img_y_start, img_y_end);
-    float end = std::max(img_y_start, img_y_end);
-    float a = lane_objects_->at(k).img_curve.a;
-    float b = lane_objects_->at(k).img_curve.b;
-    float c = lane_objects_->at(k).img_curve.c;
-    float d = lane_objects_->at(k).img_curve.d;
-
-    for (float l = start; l <= end; l++) {
-      cv::circle(*image_mat,
-                 cv::Point(static_cast<int>(GetPolyValue(a, b, c, d, l)),
-                           static_cast<int>(l)),
-                 2, lane_object_color, -1);
+      float img_y_end =
+        static_cast<float>(lane_objects_->at(k).img_curve.x_end);
+      float start = std::min(img_y_start, img_y_end);
+      float end = std::max(img_y_start, img_y_end);
+      float a = lane_objects_->at(k).img_curve.a;
+      float b = lane_objects_->at(k).img_curve.b;
+      float c = lane_objects_->at(k).img_curve.c;
+      float d = lane_objects_->at(k).img_curve.d;
+
+      for (float l = start; l <= end; l++) {
+        cv::circle(*image_mat,
+                   cv::Point(static_cast<int>(GetPolyValue(a, b, c, d, l)),
+                             static_cast<int>(l)),
+                             2, lane_object_color, -1);
+      }
+    } else {
+      CalibrationConfigManager* calibration_config_manager =
```

Comment:
Please get this homography at render() function and save globally (e.g. H_car2camera_ ) and use here. Other's may use the homography other than lane line for each frame.